server/src/main/java/org/elasticsearch/action/admin/indices/create/AutoCreateAction.java

```diff
@@ -283,7 +283,7 @@ ClusterState execute(
 
                     if (isManagedSystemIndex) {
                         final SystemIndexDescriptor descriptor = mainDescriptor.getDescriptorCompatibleWith(
-                            currentState.nodes().getSmallestNonClientNodeVersion()
```

Comment:
Is there a logic change between the smallest _non-client_ node version, and the min system index mapping version across all nodes?

Reply:
In practice, I doubt it, because only the security index uses a prior index descriptor, and it hasn't changed in a long time.

I was assuming that "non-client" was a vestige from back when we had a transport client that could show up in the cluster, but looking at how `getSmallestNonClientNodeVersion()` is created, it means data nodes or the master node. So I guess in a situation where we have a non-data ingest node or coordinating node that has an older system index mapping, but wouldn't hold data, the new code would be too conservative and create the system index with the old mappings, and update the mappings once all nodes in the cluster catch up.

I'll see if there's a simple way to preserve the current behavior.

Reply:
I committed some code that preserves the existing logic, but as I think about it, I'm not sure it's the best idea. The case where this could make a difference is when there's some non-data, non-master node on version X, but the master node and data nodes are all on version X+1. Using a smallest non-client node version lets the cluster upgrade the mappings without waiting for the version X node to upgrade, but if I understand things correctly, nothing would prevent a data node of version X from joining the mixed cluster, while the non-data/master node is still around.

I think we might want to change the logic to use the min system index mapping version across all nodes.

Reply:
Yeah, that's a good point. Using the version across all nodes seems like the most sensible option here.

Reply:
Thanks. I've undone the change.